src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -992,49 +1008,128 @@ public static void onWorldLoad(@NotNull final World world)
         }
     }
 
-    public static boolean backupColonyData()
+    /**
+     * Initiates a timer to schedule the colony backup task
+     * backupDelay of 0 disables automatic backup
+     */
+    public static void scheduleColonyBackup()
     {
-        if (numWorldsLoaded > 0 && saveNeeded)
+        final int backupDelay = Configurations.administration.colonyBackupTimeMins;
+        if (backupDelay > 0)
         {
-            saveColonies(false);
+            backupTimerTask = new backupColonyTimerTask();
+            backupTimer.schedule(backupTimerTask, backupDelay * 60 * 1000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 1000 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this multiplication operation to a "long". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2184)